graveler/sstable/cache.go

```diff
@@ -53,6 +56,12 @@ func (i *item) Close() error {
 }
 
 func NewCache(p lru.ParamsWithDisposal, fs pyramid.FS, readerOptions sstable.ReaderOptions) Cache {
+	if readerOptions.Cache == nil {
+		// Shared cache between *all* readers, otherwise sstable.NewReader keeps its
+		// refcnt but forgets only reference.  *This* cache is held inside opener so
+		// can never be finalized.
+		readerOptions.Cache = pebble.NewCache(DefaultPebbleSSTableCacheSize)
```

Comment:
Just checking - currently, there is no code that calls NewCache without sst cache in the readerOptions, right?
The shared cache in the comment is created in the EntryCatalog.

Reply:
Yes (we do inject openers in tests, calling `NewCacheWithOpener` directly... and not a pebble SSTable in sight).